hyperdash/experiment.py

```diff
@@ -105,17 +105,31 @@ def run():
         exp_thread = threading.Thread(target=run)
         exp_thread.daemon = True
         exp_thread.start()
+        self._ended = False
 
     def metric(self, name, value, log=True):
+        if self._ended:
```

Comment:
Maybe this isn't a big deal, but this could print 1000's of times

Reply:
also in the hd_client we use a logger so we can emit .info() or .warn(). Might be nice to do the same here

Reply:
Good call on the logger. 

I think it's fine to over-log as opposed to introducing more complexity via state tracking.